File: quickSort.js
const quickSort = (arr) => {
  if (arr.length < 2) return arr;
  const pivot = arr[0];
  const less = arr.filter((item) => item < pivot);
  const greater = arr.filter((item) => item > pivot);

  return quickSort(less).concat([pivot]).concat(greater);
};

const arr = [30, 15, 10, 40];
console.log(quickSort(arr));

File: quickSort.py
def quikcSort(arr):
    if len(arr) < 2:
        return arr
    pivot = arr[0]
    less = [i for i in arr[1:] if i < pivot]
    greater = [i for i in arr[1:] if i > pivot]
    
    return quikcSort(less) + [pivot] + quikcSort(greater)

arr = [30, 15, 10, 40]
print(quikcSort(arr))